python/pyspark/sql/tests/connect/test_connect_basic.py

```diff
@@ -722,6 +722,19 @@ def test_write_operations(self):
         ndf = self.connect.read.table("parquet_test")
         self.assertEqual(set(df.collect()), set(ndf.collect()))
 
+    def test_agg_with_avg(self):
+        """SPARK-41325: groupby.avg()"""
+        df = (
+            self.connect.range(10)
+            .groupBy((col("id") % lit(2)).alias("moded"))
```

Comment:
Hm, actually shouldn't we always use unresolved alias? Seems like the alias here is resolved (see https://github.com/apache/spark/commit/0f7eaeee6445aaf05310229bdec22f6953113f2e) cc @cloud-fan .

Reply:
DataFrame API creates resolved `Alias` as well. I'm not sure why `UnresolvedAlias(Alias...)` is a problem.